src/wp-includes/rest-api/endpoints/class-wp-rest-plugins-controller.php

```diff
@@ -380,25 +392,27 @@ public function create_item( $request ) {
 		/** This filter is documented in wp-includes/update.php */
 		$installed_locales = apply_filters( 'plugins_update_check_locales', $installed_locales );
 
-		$language_packs = array_map(
-			static function( $item ) {
-				return (object) $item;
-			},
-			$api->language_packs
-		);
+		if ( isset( $api ) ) {
```

Comment:
I would instead initialize `$language_packs` to an empty array before the branch at the top of the function. Then you can set it's value in the same slug branch after we've determined the response isn't a `WP_Error`.

Reply:
Yeah, that's nicer.
https://github.com/WordPress/wordpress-develop/pull/3515/commits/f2bec388bfa2f8f039f9c12b4db6893fc3783de6

Reply:
While adding the validate_callback, I realised a valid case where both slug and url are present is for URLs pointing to wordpress.org. E.g. you can install an older version of a plugin by using the URL, and still want the language pack installed from the repo.